scriptworker/ed25519.py

```diff
@@ -0,0 +1,183 @@
+
+
+def ed25519_public_key_from_string(string):
+    """Create an ed25519 public key from ``string``, which is a seed.
+
+    Args:
+        string (str): the string to use as a seed.
+
+    Returns:
+        Ed25519PublicKey: the public key
+
+    """
+    try:
+        return Ed25519PublicKey.from_public_bytes(
+            base64.b64decode(string)
+        )
+    except (UnsupportedAlgorithm, Base64Error) as exc:
+        raise ScriptWorkerEd25519Error("Can't create Ed25519PublicKey: {}!".format(str(exc)))
+
+
+def _ed25519_key_from_file(fn, path):
+    """Create an ed25519 key from the contents of ``path``.
+
+    ``path`` is a filepath containing a base64-encoded ed25519 key seed.
+
+    Args:
+        fn (callable): the function to call with the contents from ``path``
+        path (str): the file path to the base64-encoded key seed.
+
+    Returns:
+        obj: the appropriate key type from ``path``
+
+    Raises:
+        ScriptWorkerEd25519Error
+
+    """
+    try:
+        return fn(read_from_file(path, exception=ScriptWorkerEd25519Error))
+    except ScriptWorkerException as exc:
+        raise ScriptWorkerEd25519Error("Failed calling {} for {}: {}!".format(fn, path, str(exc)))
+
+
+ed25519_private_key_from_file = functools.partial(_ed25519_key_from_file, ed25519_private_key_from_string)
+ed25519_public_key_from_file = functools.partial(_ed25519_key_from_file, ed25519_public_key_from_string)
+
+
+def ed25519_private_key_to_string(key):
+    """Convert an ed25519 private key to a base64-encoded string.
+
+    Args:
+        key (Ed25519PrivateKey): the key to write to the file.
+
+    Returns:
+        str: the key representation as a str
+
+    """
+    return base64.b64encode(key.private_bytes(
+        encoding=serialization.Encoding.Raw,
+        format=serialization.PrivateFormat.Raw,
+        encryption_algorithm=serialization.NoEncryption()
+    ), None).decode('utf-8')
+
+
+def ed25519_public_key_to_string(key):
+    """Convert an ed25519 public key to a base64-encoded string.
+
+    Args:
+        key (Ed25519PublicKey): the key to write to the file.
+
+    Returns:
+        str: the key representation as a str
+
+    """
+    return base64.b64encode(key.public_bytes(
+        encoding=serialization.Encoding.Raw,
+        format=serialization.PublicFormat.Raw,
+    ), None).decode('utf-8')
+
+
+def verify_ed25519_signature(public_key, contents, signature, message):
+    """Verify that ``signature`` comes from ``public_key`` and ``contents``.
+
+    Args:
+        public_key (Ed25519PublicKey): the key to verify the signature
+        contents (bytes): the contents that was signed
+        signature (bytes): the signature to verify
+        message (str): the error message to raise.
+
+    Raises:
+        ScriptWorkerEd25519Error: on failure
+
+    """
+    try:
+        public_key.verify(signature, contents)
+    except (UnsupportedAlgorithm, InvalidSignature) as exc:
```

Comment:
I don't think `UnsupportedAlgorithm` is possible here.

Reply:
Removed.